### PR TITLE
state/meterstatus: Wired up meter status with metrics manager status

### PR DESCRIPTION
**Pull request by mattyw**
(Review request: http://reviews.vapour.ws/r/1005/)

### CODE REVIEW

File: state/metricsmanager.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			coll, closer := st.getCollection(metricsManagerC)
+			defer closer()
+			n, err := coll.FindId(st.docID(metricsManagerKey)).Count()
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if n > 0 {
+				return nil, errors.NotFoundf("metrics manager")
```
Comment (axw) [inline state/metricsmanager.go]: This seems like the wrong error to return. It's the opposite of what's true.

File: state/metricsmanager.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+// GracePeriod returns the current grace period.
+func (m *MetricsManager) GracePeriod() time.Duration {
+	return m.doc.GracePeriod
+}
+
+// MetricsManager returns an existing metricsmanager, or a new one if non exists.
+func (st *State) MetricsManager() (*MetricsManager, error) {
+	mm, err := st.getMetricsManager()
+	if errors.IsNotFound(err) {
```
Comment (axw) [inline state/metricsmanager.go]: I think you should fold `newMetricsManager` in here, and change the buildTxn around a bit to do:

loop:
    mm, err := st.getMetricsManager()
    if !errors.IsNotFound(err) {
        return mm, err
    }
    return op that creates if not exists

File: state/metricsmanager.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+func (m *MetricsManager) gracePeriodExceeded() bool {
+	now := time.Now()
+	t := m.LastSuccessfulSend().Add(m.GracePeriod())
+	return t.Before(now) || t.Equal(now)
+}
+
+// MeterStatus returns the overall state of the MetricsManager as a meter status summary.
+func (m *MetricsManager) MeterStatus() (code, info string) {
```
Comment (axw) [inline state/metricsmanager.go]: why not return MeterStatusCode here?

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Pull request merged**